### PR TITLE
Basic Github Actions workflow

### LINKED ISSUE #1: Configure Git Actions for project
configure actions for build + ut

@kawapa
@emttiew

### LINKED ISSUE #4: Configuring CMake
add below libraries to CMake:

-GTEST **[DONE]**
-BOOST
-OPEN SSL
-REST API LIBRARY (do research and select some lib for rest api)

try to use FetchContent module to let every library to be downloaded automatically from repository

@Kacperfis @BarTes8 @emttiew

### PR DESCRIPTION
**Pull request by emttiew**
### Issues
#1 #4 

### Description

1. Add basic file structure
2. Add basic CMake config with Gtest
3. Add worflow action that triggers on push to main or pull request

Provided action will attempt to build and run unit tests when triggered.

### CODE REVIEW

**Comment by ziobron** [comment]
@emttiew, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

File: .github/workflows/cmake-ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Buld & Test on push/pull_request
```
Comment (Kacperfis) [inline .github/workflows/cmake-ci.yml]: Build

Comment (ziobron) [inline .github/workflows/cmake-ci.yml]: This is name for the whole CI pipeline, so I propose to name it "Password Manager CI" or something simillar.

File: .github/workflows/cmake-ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+on:
+  push:
+    branches:
+        - 'main'
+  pull_request:
+    types: [opened, reopened, edited]
```
Comment (ziobron) [inline .github/workflows/cmake-ci.yml]:
```suggestion
on: [push, pull_request, workflow_dispatch]
```

File: .github/workflows/cmake-ci.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    - name: Setup cmake
+      uses: jwlawson/actions-setup-cmake@v1.14
+      with:
+        cmake-version: '3.22.1'
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
```
Comment (ziobron) [inline .github/workflows/cmake-ci.yml]: It's good to create variables for things that are used multiple times, like the build directory. Thanks to that you need to change them only in one place.

```
env:
      BUILD_PATH: ${{github.workspace}}/build
```

```suggestion
      run: cmake -B ${{env.BUILD_PATH}} -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
```

**Review by ziobron — APPROVED** [review]
Also fix missing new lines at the end of files

File: .github/workflows/cmake-ci.yml (hunk 2 above)
Comment (kawapa) [reply .github/workflows/cmake-ci.yml]: Worth to point out that 'workflow_dispatch' will allow you to trigger action manually from the GH website.

**Comment by ziobron** [comment]
Thanks @kawapa for Code Review.
🏅 1 XP granted

File: .github/workflows/cmake-ci.yml (hunk 2 above)
Comment (emttiew) [reply .github/workflows/cmake-ci.yml]: I was thinking about making the manual version of the job a separate action. What do you think?

Comment (emttiew) [reply .github/workflows/cmake-ci.yml]: Maybe for now I will set this job with `workflow_dishpatch`. We can think about improvements later.

**Review by Kacperfis — APPROVED** [review]

**Pull request merged**